src/index.js

```diff
@@ -136,23 +137,25 @@ function getOptionModel( state, envelope ) {
 }
 
 function getFullOptionModel( state ) {
-	var version = 1;
-	if ( !state.fullOptionModels[ version ] ) {
-		state.fullOptionModels[ version ] = HyperResource.routesGenerator( state.resources, state.prefix, version )( state.engines );
+	var maxVersion = getMaxVersion( state );
+	for( var i = 1; i <= maxVersion; i ++ ) {
+		if ( !state.fullOptionModels[ i ] ) {
+			state.fullOptionModels[ i ] = HyperResource.routesGenerator( state.resources, state.prefix, i )( state.engines );
+		}
 	}
-	return state.fullOptionModels[ version ];
+	return state.fullOptionModels;
 }
 
 function getMaxVersion( state ) { // jshint ignore:line
 	return _.reduce( state.resources, function( version, resource ) {
-		var max = _.maxBy( _.keys( resource.versions ), parseInt );
+		var max = parseInt( _.maxBy( _.keys( resource.versions ), parseInt ) ) || 0;
```

Comment:
@arobson: Yeah, you'd want to use `_.unary( parseInt )` or something similar. There's nothing wrong with what is already there but I saw this line, remembered you making some sort of comment about pain and `maxBy`, and thought it might be cleaner to convert the keys to integers once.